core/src/main/scala/org/apache/spark/status/protobuf/JobExecutionStatusSerializer.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.status.protobuf
+
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.spark.JobExecutionStatus
+
+object JobExecutionStatusSerializer {
+
+  private def PREFIX = "JOB_EXECUTION_STATUS_"
+
+  private[protobuf] def serialize(input: JobExecutionStatus): StoreTypes.JobExecutionStatus = {
+    StoreTypes.JobExecutionStatus.valueOf(PREFIX + input.toString)
```

Comment:
How much additional delay will string join and string remove bring?

Reply:
or following as？
<img width="783" alt="image" src="https://user-images.githubusercontent.com/15246973/210049089-bba233ff-e982-48f5-8a7e-8488c7cc2488.png">

Reply:
how about use a map to store the mapping relationships?

Reply:
+1 for using pattern match here.

Reply:
> @panbingkun shall we refactor `enum DeterministicLevel` as well?

Done